### PR TITLE
Revamp dataplane client

### PR DESCRIPTION
**Pull request by Chuhaa**
# Description
Revamp dataplane client

Fixes # ([issue](https://github.com/ballerina-platform/ballerina-extended-library/issues/201))

One line release note: 
- Revamp dataplane client to support target type based return type.

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit test

**Test Configuration**:
* Ballerina Version: SL Beta6
* Java SDK: 11

# Checklist:

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: cosmos-native/src/main/java/io/ballerinax/cosmosdb/DataplaneClient.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    public static Object close() {
+        try {
+            cosmosClient.close();
+            return null;
+        } catch (Exception e) {
+            return BallerinaErrorGenerator.createBallerinaDatabaseError(e);
+        }
+    }
+
+    private static void setCustomConfiguration(Object customConfig) {
```
[the remark this answers is not in the document]
Comment (Chuhaa) [reply cosmos-native/src/main/java/io/ballerinax/cosmosdb/DataplaneClient.java]: Moved to Utils.bal. Thanks

File: cosmos-native/build.gradle [hunk 2]
```diff
@@ -0,0 +1,31 @@
+plugins {
+    id 'java'
+    id "com.github.johnrengelman.shadow"
+}
+
+group project.group
```
[the remark this answers is not in the document]
Comment (Chuhaa) [reply cosmos-native/build.gradle:6]: Fixed in https://github.com/ballerina-platform/module-ballerinax-azure-cosmosdb/commit/896a6fefee6c8f6a6f20d4fc1474feba16cbd062

**Review by RolandHewage — APPROVED** [review]

File: cosmosdb/client_endpoint.bal [hunk 3]
```diff
@@ -37,321 +38,232 @@ public isolated client class DataPlaneClient {
     #
     # + cosmosdbConfig - Configurations required to initialize the `Client` endpoint
     # + httpClientConfig - HTTP configuration
-    # + return -  Error at failure of client initialization
-    public isolated function init(ConnectionConfig cosmosdbConfig, http:ClientConfiguration httpClientConfig = {}) returns Error? {
+    # + return - Error at failure of client initialization
+    public isolated function init(ConnectionConfig cosmosdbConfig, http:ClientConfiguration httpClientConfig = {},
+    CustomizedConfiguration? customConfig = ()) returns Error? {
```
Comment (abeykoon) [inline cosmosdb/client_endpoint.bal:43]: Shall we remove ClientConfiguration as we are using the SDK?

Comment (abeykoon) [reply cosmosdb/client_endpoint.bal:43]: cosmosdbConfig --> connectionConfig
customConfig --> advanceClientConfig (Type needs to be ClientConfiguration)

File: cosmosdb/client_endpoint.bal [hunk 4]
```diff
@@ -37,321 +38,232 @@ public isolated client class DataPlaneClient {
...
+    #
     # + databaseId - ID of the database to which the container belongs to
     # + containerId - ID of the container where, document is created
+    # + documemtId - ID of the document
     # + document - A JSON document to be saved in the database
     # + partitionKey - The specific value related to the partition key field of the container 
-    # + documentCreateOptions - The `cosmos_db:DocumentCreateOptions` which can be used to add additional capabilities 
-    #                           to the request
-    # + return - If successful, returns `cosmos_db:Document`. Else returns `cosmos_db:Error`.
+    # + requestOptions - The `cosmos_db:DocumentCreateOptions` which can be used to add additional capabilities
```
Comment (abeykoon) [inline cosmosdb/client_endpoint.bal:58]: Improve the description that we can override what we set at init

Comment (Chuhaa) [reply cosmosdb/client_endpoint.bal]: Resolved in https://github.com/ballerina-platform/module-ballerinax-azure-cosmosdb/commit/e8efdeb8fbaf0bcf88e0b231605b1dc7df3b858e

**Review by abeykoon — APPROVED** [review]

**Pull request merged**